### DISCUSSION TITLE
Inserting streaming data using python api tbapi

### DISCUSSION THREAD

**Discussion by danielvaz21** [finos/TimeBase-CE#284] [General]
Hi @alex-karpovich,

We have been using the function InstrumentMessage from tbapi to insert data into a space of a stream in timebase from what we understand is that it inserts one records at a time. Is there a way to insert multiple records at a time.

Regards
Daniel Vaz

**Comment by alex-karpovich** [comment]
Greetings, Daniel

Please take a look at the samples here: https://github.com/epam/TimeBaseTutorial/blob/main/python/samples/write_stream.py

we do not have `batch` API, because it will be same execution internally as using loop and execute `send()`
